### ISSUE TITLE
Local LAN DNS - Inconsistent Behavior

### ISSUE THREAD

**Issue by dom-madrid** [pi-hole/pi-hole#3753]
### Versions
  Pi-hole version is v5.1.2 (Latest: v5.1.2)
  AdminLTE version is v5.1.1 (Latest: v5.1.1)
  FTL version is v5.2 (Latest: v5.2)

### Platform
- OS and version:  Raspberry OS v10 and Ubuntu 18.04
- Platform:  Raspberry Pi, Virtualbox 6 (Ubuntu 18.04), Bare Metal (Ubuntu 18.04)

### Expected behavior

With the Pi-Hole configured for DHCP, DNS and Ad Blocking.
Lan DNS should return local IP address for local servers and Ads should be blocked

### Actual behavior / bug

Different results by client OS.... 
On Windows 10 Desktop (Wired) / Laptop (WIFI) : 
DHCP OK.
DNS for LAN Servers: Resolve to public IP
AD not Blocked

On Ubuntu Desktop (Wired) / Laptop (WIFI)
DHCP OK.
DNS for LAN Servers: Resolve to local IP for about 5 minutes then resolve to public IP
AD Blocked

On Android 9/10 Smartphone (WIFI)
DHCP OK.
on Rpi install: DNS for LAN Servers: Resolve to local IP as expected.
on VM install: DNS for LAN Servers: Resolve to local IP for about 5 minutes then resolve to public IP
AD Blocked

On Android 8 Tablet (WIFI)
DHCP OK.
on Rpi install: DNS for LAN Servers: Resolve to local IP as expected.
AD Blocked

On IPhone IOS8 (WIFI)
DHCP OK.
on Rpi install: DNS for LAN Servers: Resolve to local IP as expected.
AD Blocked

### Steps to reproduce

1. Install pihole 5
2. Configure LAN DNS
3. Connect to the LAN / WAN

## Debug Token

- URL:  https://tricorder.pi-hole.net/ksajqy55d1

### Additional context

Although I can reproduce it anymore, I tend to remember the process working as expected under all conditions with Pihole V4.

**Comment by yubiuser** [comment]
It looks like a client issue at a first glance. You're clients do get public Ipv6 addresses and thereby may circumvent pihole.
```
IPv6 address(es) bound to the eth0 interface:
   2a01:XXXXXXXXX does not match the IP found in /etc/pihole/setupVars.conf (https://discourse.pi-hole.net/t/use-ipv6-ula-addresses-for-pi-hole/2127)
```

**Comment by rchenzheng** [comment]
Did the machine with the problem get the pihole DNS?

**Comment by dom-madrid** [comment]
@yubiuser 
Thanks for catching that one. 
@rchenzheng 
So far I tested only with the Raspberry setup. I have to go and test on a different site to test the Ubuntu Server Setup.
On the rPi setup, for Android, IOS, WIN7 and Linux machines, the LAN DNS are coming correctly from PiHole. No WIN10 around for testing but I guess it should work as well.

Now the problem is that the various ISP routers do not behave in the same way. I one case, I just had to disable IPv6 DHCP in the router and let Pi-Hole do the job. In another router, I am currently stuck, as no IP6 options are user accessible, so although I disabled the DHCP and DNS part, it seems to only affect the IP4 part, which complicates matters as the router does not offer loopback for local addresses... Any idea for that one are welcomed. I have to test other setups with other ISPs but I guess they will fall in either one of the cases described before.
